docs_src/design/adr/013-Device-Service-Events-Message-Bus.md

```diff
@@ -0,0 +1,259 @@
+
+Services receiving the Event DTO from the Message Bus will log validation errors and stop processing the Event.
+
+### Message Envelope
+
+EdgeX Go Services currently uses a custom Message Envelope for all data that is published to the Message Bus. This envelope wraps the data with metadata, which is `ContentType` (JSON or CBOR), `Correlation-Id` and the obsolete `Checksum`. The `Checksum` is used when the data is CBOR encoded to identify the Event in V1 API to be mark it as pushed. This checksum is no longer needed as the V2 Event DTO requires the ID be set by the Device Services which will always be used in the V2 API to mark the Events as pushed. The Message Envelope will be updated to remove this property.
+
+The C SDK will recreate this Message Envelope.
+
+### Application Services
+
+As part of the V2 API consumption work in Ireland the App Services SDK will be changed to expect to receive V2 Event DTOs rather than the V1 Event model. It will also be updated to no longer expect or use the `Checksum` currently on the  Message Envelope. Note these changes must occur for the V2 consumption and are not directly tied to this effort. 
+
+The App Service SDK will be enhanced for the secure Message Bus connection described below. See **[Secure Connections](#secure-connections)** for details
+
+### Message Bus Topics
+
+> *Note: The change recommended here is not required for this design, but it provides a good opportunity to adopt it.*
+
+Currently Core Data publishes Events to the simple `events` topic. All Application Services running receive every Event published, whether they want them or not. The Events can be filtered out using the `FilterByDeviceName` pipeline function, but the Application Services still receives every Event and process all the Events to some extent. This could cause load issues in a deployment with many devices and large volume of Events from various devices or a very verbose device that that Application Services is not interested in.
+
+> *Note: The current `FilterByDeviceName` is only good if the device name is known statically and the only instance of the device defined by the `DeviceProfileName`. What we really need is `FilterByDeviceProfileName` which allows multiple instances of a device to be filtered for, rather than a single instance as it it now. The V2 API will be adding `DeviceProfileName` to the Events, so in Ireland this  filter will be possible.*
+
+Pub/Sub systems have advanced topic schema, which we can take advantage of from Application Services to filter for just the Events the Application Service actual wants. Publishers of Events must add the `DeviceProfileName` and/or `DeviceName` to the topic in the form `edgex/events/<device-profile-name>/<device-name>`. This allows Application Services to filter for just the Events from the device(s) it wants by only subscribing to those `DeviceProfileNames` or the specific `DeviceNames`. Example subscribe topics if above schema is used:
+
+- **edgex/events/#**
```

Comment:
What do you think about using "device-events" vs. "events"?

Reply:
Why? Are there other Events that are not device events? metrics??

Reply:
BTW, these are examples of how the configuration could be set. If one wanted, they code use `device-events` rather than `events`.